File: src/main/scala/com/exasol/cloudetl/filesystem/HiddenFilesFilter.scala
package com.exasol.cloudetl.filesystem

import org.apache.hadoop.fs.Path
import org.apache.hadoop.fs.PathFilter

object HiddenFilesFilter extends PathFilter {

  private[this] val PREFIX_DOT = "."
  private[this] val PREFIX_UNDERSCORE = "_"

  override def accept(p: Path): Boolean = {
    val name = p.getName()
    !name.startsWith(PREFIX_DOT) && !name.startsWith(PREFIX_UNDERSCORE)
  }

}

File: src/main/java/com/exasol/cloudetl/filesystem/HiddenFilesFilter.java
package com.exasol.cloudetl.filesystem;

import org.apache.hadoop.fs.Path;
import org.apache.hadoop.fs.PathFilter;

/** Filters hidden filesystem entries. */
public final class HiddenFilesFilter implements PathFilter {
    /** Singleton instance. */
    public static final HiddenFilesFilter INSTANCE = new HiddenFilesFilter();

    private HiddenFilesFilter() {
        // Singleton.
    }

    @Override
    public boolean accept(final Path path) {
        final String name = path.getName();
        return !name.startsWith(".") && !name.startsWith("_");
    }
}
